[DOCS] Adds cohere service to inference API docs

szabosteve opened the pull request:
## Overview

This PR expands the PUT inference API docs with the `cohere` service parameters and example.

### Preview

[PUT inference API docs](https://elasticsearch_bk_105394.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-inference-api.html)

szabosteve committed:
[DOCS] Adds cohere service to inference API docs.

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

jonathan-buttner commented on docs/reference/inference/put-inference.asciidoc:
```diff
@@ -64,16 +65,47 @@ The type of the {infer} task that the model will perform. Available task types:
 (Required, string)
 The type of service supported for the specified task type.
 Available services:
+* `cohere`: 
```
How about we add:

```
specify the `text_embedding` task type to use the Cohere service.
```

jonathan-buttner commented on docs/reference/inference/put-inference.asciidoc:
```diff
@@ -64,16 +65,47 @@ The type of the {infer} task that the model will perform. Available task types:
+the same name and the updated API key.
+
+`embedding_type`::
+(Required, string)
```
Let's switch this to `Optional`. We can say it defaults to `float` if not specified: https://docs.cohere.com/reference/embed under `embedding_types`

jonathan-buttner commented on docs/reference/inference/put-inference.asciidoc:
```diff
@@ -64,16 +65,47 @@ The type of the {infer} task that the model will perform. Available task types:
+  * `float`: use it for the default float embeddings.
+  * `int8`: use it for signed int8 embeddings.
+
+`model_id`::
```
👍

jonathan-buttner commented on docs/reference/inference/put-inference.asciidoc:
```diff
@@ -141,11 +173,31 @@ Settings to configure the {infer} task. These settings are specific to the
+For `cohere` service only. Specifies the type of input passed to the model.
+Valid values are:
+  * `ingest`: use it for storing document embeddings in a vector database.
+  * `search`: use it for stroing embeddings of search queries run against a 
```
```suggestion
  * `search`: use it for storing embeddings of search queries run against a 
```

jonathan-buttner commented on docs/reference/inference/put-inference.asciidoc:
```diff
@@ -141,11 +173,31 @@ Settings to configure the {infer} task. These settings are specific to the
+`input_type`:::
+(optional, string)
+For `cohere` service only. Specifies the type of input passed to the model.
+Valid values are:
```
I also added

- `"classification"`: Used for embeddings passed through a text classifier.
- `"clustering"`: Used for the embeddings run through a clustering algorithm.

jonathan-buttner reviewed (commented):
Looks good. Just an idea but as we support more 3rd party services I wonder if it'll make sense to create like a section for each so users can go directly to that section instead of having to look at each field to determine if it is applicable for the service they're trying to configure.

Something like a heading for each service and within the its section the specific fields to configure that servce.

szabosteve committed:
[DOCS] Adds description to service param.

szabosteve committed:
[DOCS] Addresses feedback.

szabosteve commented:
@jonathan-buttner Thank you for the review. I addressed your feedback via [ead1be8](https://github.com/elastic/elasticsearch/pull/105394/commits/ead1be8482fcb051d1a4e59a1326b82fe47bd940). Restructuring the page by service type is a great idea. I'll think about the best way to do it and cover it in a separate PR if it's okay. I need to open another one for the built-in and Eland models anyway.

jonathan-buttner reviewed (approved):
Thanks for the changes!
